great_expectations/self_check/util.py

```diff
@@ -467,6 +467,24 @@
 # except ImportError:
 #     teradatasqlalchemy = None
 
+try:
+    from great_expectations.dataset import SqlAlchemyDataset
+    from great_expectations.dataset.sqlalchemy_dataset import SqlAlchemyBatchReference
```

Comment:
Why do we need to try/except imports from our own library? Because SQL Alchemy might not be installed? Can we use [this pattern](https://github.com/great-expectations/great_expectations/pull/6731#discussion_r1064903709)?

Reply:
@NathanFarmer Yes -- this is exactly for the case for when the corresponding Spark and SQLAlchemy imports are not present.  Is saying `from __future__ import __annotations__` at the top enough for solving this problem?

Reply:
@Kilo59 Yes, in the branch for Multi-Batch, it breaks tests.  This is because the multi-batch implementation is less forgiving to sloppy code.